src/main/java/nl/tudelft/planningstool/api/v1/UserInfoAPI.java

```diff
@@ -65,6 +67,21 @@ public UserCoursesResponse get(@PathParam("userId") String userId) {
                 .collect(Collectors.toList());
     }
 
+    @GET
+    @Path("/assignmentsAsTeacher")
+    public List<AssignmentResponse> getAssignmentsWhereTeacher(@PathParam("userId") String userId) {
+        return this.userDAO.getFromUUID(userId)
+                .getCourses().stream()
+                .filter(courseRelation -> courseRelation.getCourseRole() == CourseRelation.CourseRole.TEACHER)
+                .map(CourseRelation::getCourse)
+                .map(Course::getAssignments)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toList())
+                .stream()
```

Comment:
These two lines are unneccessary. You originally already had a stream, so no need to recreate it.